### PR TITLE
Don't add rpath to swift with statically linking

### PR DESCRIPTION
**Pull request by keith**
This updates the rpath linking logic to only add the rpath that points
to the swift dylibs in the case that the libraries are not statically
linked into the binary.

This fix is marginally relevant to https://bugs.swift.org/browse/SR-1967, but does not solve the core problem here.

### CODE REVIEW

**Comment by dduan** [comment]
@swift-ci please test

File: lib/Driver/ToolChains.cpp [hunk 1]
```diff
@@ -1283,10 +1284,10 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
     Arguments.push_back(context.Args.MakeArgString(LibProfile));
   }
 
-  // FIXME: We probably shouldn't be adding an rpath here unless we know ahead
-  // of time the standard library won't be copied.
```
Comment (jrose-apple) [inline lib/Driver/ToolChains.cpp]: Please preserve this FIXME in the new code as well—it's referencing the rest of SR-1967.

**Review by jrose-apple — COMMENTED** [review]
Thanks for picking this up, Keith! Can you add a test to test/Driver/linker.swift as well?

File: lib/Driver/ToolChains.cpp (hunk 1 above)
Comment (keith) [inline lib/Driver/ToolChains.cpp]: Done, also moved this up into the else above

**Comment by keith** [comment]
Yep I'll look at adding that test!

**Comment by keith** [comment]
(sorry about the tests, the first push here was bogus)

**Comment by CodaFi** [comment]
@swift-ci please test

**Comment by CodaFi** [comment]
@swift-ci please test

**Comment by keith** [comment]
I've just pushed a test for this. I would love some feedback on how I'm using `FileCheck` to validated the absence of `-rpath`. AFACIT there's no good way to do this. Really I want something like:

```swift
// SWIFT_STATIC-DAG-NOT: -rpath
```

Instead I'm using the `-implicit-check-not` flag, which works in this case it seems, but requires a more ordered list of arguments, rather than ignoring the order and using `-DAG` for all of these (since the order doesn't actually matter)

**Comment by dduan** [comment]
@swift-ci please test

**Comment by dduan** [comment]
@swift-ci please test

**Comment by keith** [comment]
@swift-ci please test

**Comment by dduan** [comment]
@swift-ci please test

**Comment by keith** [comment]
Looks like these failures are because I force pushed, it's still building the new sha.

**Comment by dduan** [comment]
@swift-ci please test

**Comment by keith** [comment]
(it seems to still be running)

**Comment by dduan** [comment]
@swift-ci please smoke test OS X Platform

**Review by jrose-apple — APPROVED** [review]

**Comment by keith** [comment]
Thanks!

**Pull request merged**